mack/__init__.py

```diff
@@ -384,6 +384,33 @@ def find_composite_key_candidates(
                 return list(df_col_excluded.select(*c).columns)
 
 
+def find_all_composite_key_combos(
+    df: Union[DeltaTable, DataFrame], exclude_cols: List[str] = None
+):
+    if type(df) == DeltaTable:
+        df = df.toDF()
+    if exclude_cols is None:
+        exclude_cols = []
+    df_col_excluded = df.drop(*exclude_cols)
+    col_select_condition = df_col_excluded.distinct().count()
```

Comment:
@souvik-databricks why does it need to distinct? If .distinct().count() < .count(), there is no unique combination, or?

Reply:
@robertkossendey .distinct() is required because if your data has duplicates in them and then the collect_set() in line 401 will matchup and we have  only unique combination of values for the columns.

Reply:
@souvik-databricks still struggling to understand, could you add a test with a completely duplicate line? Because that should return no combination at all right?

Reply:
@robertkossendey So in that scenario if you have a case of duplicate record then that would indicate that from the columns that you have included to search in, you don't have valid composite key combos. You would have to include more columns to act as part of the composite key.

If in the whole dataset you have a full exact duplicate record then it won't make any sense to keep it as that's then redundant information. 

And I will add a test with duplicate record with the explanation of the output for that test case in the readme.

Reply:
> If in the whole dataset you have a full exact duplicate record then it won't make any sense to keep it as that's then redundant information.

@souvik-databricks but isn't that the information that there is no valid key candidate in the provided dataset?

Reply:
I was OOO for last week. @robertkossendey I am back now, I will resume the work now.